theano/tensor/nnet/abstract_conv.py

```diff
@@ -718,10 +751,17 @@ def conv2d(self, img, kern, mode="valid"):
             for b in xrange(img.shape[0]):
                 for n in xrange(kern.shape[0]):
                     for im0 in xrange(img.shape[1]):
-                        # some cast generates a warning here
-                        out[b, n, ...] += _convolve2d(img[b, im0, ...],
-                                                      kern[n, im0, ...],
-                                                      1, val, bval, 0)
+                        for row_offset in xrange(dilation[0]):
+                            for col_offset in xrange(dilation[1]):
+                                fragment = img[b, im0,
+                                               row_offset::dilation[0],
+                                               col_offset::dilation[1]]
```

Comment:
No problem, it's just that we run correctness tests against that implementation, and those can take long...